ksonnet-gen/ksonnet/emit.go

```diff
@@ -46,26 +87,59 @@ type root struct {
 
 	ksonnetLibSHA *string
 	k8sSHA        *string
+
+	isLegacySchema bool
```

Comment:
Personally? I would make `usesLegacySchema` a method (_e.g._, `root#isLegacySchema`) which inspects `spec.info.Version` and returns true if the version works.

Rationale is, I think it's a little harder to maintain `isLegacySchema` as state, and since we don't care about perf here, I don't see a penalty to re-parsing and comparing versions any time we need to know whether it's a legacy schema. It also simplifies `newRoot` somewhat.